Composer/packages/client/src/pages/design/DesignPage.tsx

```diff
@@ -708,6 +810,37 @@ const DesignPage: React.FC<RouteComponentProps<{ dialogId: string; projectId: st
             onDismiss={() => dismissManifestModal(projectId)}
           />
         )}
+        {brokenSkillInfo && (
+          <RepairSkillModal
+            skillItem={brokenSkillInfo}
+            onDismiss={() => {
+              setBrokenSkillInfo(undefined);
+            }}
+            onNext={(option) => {
+              const skillIdToRemove = brokenSkillInfo.skillId;
+              if (!skillIdToRemove) return;
+
+              if (option === RepairSkillModalOptionKeys.repairSkill) {
+                setCreationFlowTypes('Skill');
+                setCreationFlowStatus(CreationFlowStatus.OPEN);
+                setBrokenSkillRepairCallback(() => {
```

Comment:
I see this callback used twice. Refactor to a function

Reply:
I did't see 'setBrokenSkillRepairCallback' used twice, can you give more detail ?